src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -290,7 +297,7 @@ protected interface ISaslMechanism
         /// <summary>
         /// Represents a SASL step.
         /// </summary>
-        protected interface ISaslStep
+        protected internal interface ISaslStep
```

Comment:
Since we need to promote `ScramShaMechanism` from `private` to  `internal`and `ScramShaMechanism` exposes `ISaslStep`, we have to promote `ISaslStep` to `protected internal`

Reply:
Would not be necessary if you adopt my suggestion above.